crates/ecs/src/lib.rs

```diff
@@ -648,6 +908,28 @@ impl World {
         archetypes
     }
 
+    fn get_archetype_with_matching_components(
+        &self,
+        component_type_ids: &Vec<TypeId>,
+    ) -> Option<ArchetypeIndex> {
+        if component_type_ids.is_empty() {
+            return Some(0);
+        }
+        // Get all archetypes containing all the component types
+        let arch_ids = self.get_archetype_indices(component_type_ids);
+        let number_of_components = component_type_ids.len();
+
+        // Find if archetype containing only relevant components exists
+        arch_ids.into_iter().find(|&arch_id| {
+            self.archetypes
+                .get(arch_id)
+                .expect("Archetype should exist since index was just fetched.")
+                .component_typeid_to_component_vec
+                .len()
+                == number_of_components
+        })
```

Comment:
Could probably add panic if multiple archetypes match the lenght since this should not be possible and would be an indication of implementation error

Reply:
Yeah, you can do that in a clean way by simply using an assert, something like `assert!(exactly_matching_archetypes == 1)`.

(side-note:) You could call the closure `only_contains_matching_components` instead of `contains_matching_components` to make it even clearer what it checks for.